raiden_contracts/data/source/services/ServiceRegistry.sol

```diff
@@ -43,16 +43,149 @@ contract Deposit {
+    function refresh_price() private {
+        set_price = current_price();
+        set_price_at = now;
+    }
+
+    function set_price_bump_parameters(
+            uint256 _price_bump_numerator,
+            uint256 _price_bump_denominator
+    ) private {
+        require(_price_bump_denominator > 0, "divide by zero");
+        price_bump_numerator = _price_bump_numerator;
+        price_bump_denominator = _price_bump_denominator;
+    }
+
+    function set_min_price(uint256 _min_price) private {
+        // No checks.  Even allowing zero.
+        min_price = _min_price;
+        // No checks or modifications on set_price.
+        // Even if set_price is smaller than min_price, current_price() function returns min_price.
+    }
+
+    function set_decay_constant(uint256 _decay_constant) private {
+        require(_decay_constant > 0, "attempt to set zero decay constant");
+        require(_decay_constant < 2 ** 60, "too big decay constant");
+        decay_constant = _decay_constant;
+    }
+
+    function set_registration_duration(uint256 _registration_duration) private {
+        // No checks.  Even allowing zero (when no new registrations are possible).
+        registration_duration = _registration_duration;
+    }
+
+
+    /// @notice The amount of deposits for registration or extension.
+    /// Note: the price moves quickly depending on what other addresses do.
+    /// The current price might change after you send a `deposit()` transaction
+    /// before the transaction is executed.
+    function current_price() public view returns (uint256) {
+        require(now >= set_price_at, "An underflow in price computation");
+        uint256 seconds_passed = now - set_price_at;
+
+        return decayed_price(set_price, seconds_passed);
+    }
+
+
+    /// @notice Calculates the decreased price after a number of seconds.
+    /// @param _set_price The initial price.
+    /// @param _seconds_passed The number of seconds passed since the initial
+    /// price was set.
+    function decayed_price(uint256 _set_price, uint256 _seconds_passed) public
+        view returns (uint256) {
+        // We are here trying to approximate some exponential decay.
+        // exp(- X / A) where
+        //   X is the number of seconds since the last price change
+        //   A is the decay constant (A = 200 days corresponds to 0.5% decrease per day)
+
+        // exp(- X / A) ~~ P / Q where
+        //   P = 24 A^4
+        //   Q = 24 A^4 + 24 A^3X + 12 A^2X^2 + 4 AX^3 + X^4
+        // Note: swap P and Q, and then think about the Taylor expansion.
+
+        uint256 X = _seconds_passed;
+
+        if (X >= 2 ** 60) { // The computation below overflows.
+            return min_price;
+        }
+
+        uint256 A = decay_constant;
+
+        uint256 P = 24 * (A ** 4);
+        uint256 Q = P + 24*(A**3)*X + 12*(A**2)*(X**2) + 4*A*(X**3) + X**4;
+
+        uint256 price = _set_price * P / Q;
```

Comment:
If both `decay_constant` and `_set_price` are very large, this will overflow. Practically, this can't be triggered by a user of the contract and won't happen unless the controller sets nonsensical values.

Reply:
I'll limit the `decay_constant` at `2 ** 60` or something.

I'm thinking whether or not to code a hard-limit on the price.  This complicates code for an unlikely scenario, where things have already gone crazy.

Reply:
Ah, the `decay_constant` is already limited at `2 ** 60`.

But `price_bump_numerator` was not capped.

The price limit of 2 ** 60 would be way too low, but 2 ** 128 looks ok.

Reply:
The overflow is closer than I thought.

I think the price cap at `2 ** 90` and the decay_constant cap at `2 ** 40` are reasonable.

Reply:
I added those limits.  I still want to add tests that try to cross the limit.